### PR TITLE
Merge TestingKafka classes

### PR DESCRIPTION
**Pull request by kokosing**
Retry sending kafka message in tests

### CODE REVIEW

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java [hunk 1]
```diff
@@ -13,7 +13,6 @@
  */
 package io.prestosql.plugin.kafka;
 
-import com.google.common.collect.ImmutableList;
```
Comment (skrzypo987) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java:16]: nit: remove one of the `only` words in the second sentence of commit message

File: presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java [hunk 2]
```diff
@@ -30,6 +33,13 @@
 
     void createTopicWithConfig(int partitions, int replication, String topic, boolean enableLogAppendTime);
 
+    default <K, V> RecordMetadata sendMessages(Stream<ProducerRecord<K, V>> recordStream)
+    {
+        return sendMessages(ImmutableMap.of(), recordStream);
+    }
+
+    <K, V> RecordMetadata sendMessages(Map<String, String> extraProducerProperties, Stream<ProducerRecord<K, V>> recordStream);
```
Comment (skrzypo987) [inline presto-testing-kafka/src/main/java/io/prestosql/testing/kafka/TestingKafka.java:41]: "Optional" arguments should be at the end of arg list.

**Review by skrzypo987 — APPROVED** [review]

**Comment by kokosing** [comment]
@skrzypo987 AC + more

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java [hunk 3]
```diff
@@ -88,60 +89,84 @@ protected QueryRunner createQueryRunner()
...
         String topic = "topic-Key-Subject";
-        assertTopic(topic,
+        assertTopic(
+                topic,
                 format("SELECT \"%s-key\", col_1, col_2 FROM %s", topic, toDoubleQuoted(topic)),
                 format("SELECT \"%s-key\", col_1, col_2, col_3 FROM %s", topic, toDoubleQuoted(topic)),
                 true,
-                () -> testingKafkaWithSchemaRegistry.createConfluentProducer());
+                ImmutableMap.<String, String>builder()
+                        .put(SCHEMA_REGISTRY_URL_CONFIG, testingKafka.getSchemaRegistryConnectString())
```
Comment (Praveen2112) [inline plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestKafkaWithConfluentSchemaRegistryMinimalFunctionality.java:114]: Can we extract the common properties ?

**Review by Praveen2112 — APPROVED** [review]

**Comment by kokosing** [comment]
Flaky https://github.com/trinodb/trino/issues/6412

**Pull request merged**